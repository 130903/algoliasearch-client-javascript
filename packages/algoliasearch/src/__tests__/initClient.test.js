// @flow

import { initClient } from '../';

const validParams = { appId: '', apiKey: '' };

it('throws when it has too little parameters', () => {
  expect(() => initClient({})).toThrow();
  expect(() => initClient({ appId: '' })).toThrow();
  expect(() => initClient({ apiKey: '' })).toThrow();
});

it('contains the correct methods', () => {
  const client = initClient(validParams);
  expect(Object.keys(client)).toMatchSnapshot();
});

it('batch', () => {
  const fakeRequest = o => o;
  const client = initClient(validParams);
  const requests = [
    client.batch(fakeRequest, [
      {
        action: 'addObject',
        indexName: 'cool-index',
        body: {
          some: 'data',
          nested: { is: { all: 'fine' } },
          function: is => `${is}also allowed, but not smart`,
        },
      },
    ]),
    client.batch(fakeRequest, [
      {
        action: 'updateObject',
        indexName: 'uncool-index',
        body: {
          objectID: 'some-cool-object',
          some: 'other data',
          and: 'this is everything now',
        },
      },
    ]),
    client.batch(fakeRequest, [
      {
        action: 'partialUpdateObject',
        indexName: 'cool-index',
        body: {
          objectID: 'some-cool-object',
          some: 'data',
          the: 'rest is still here',
        },
      },
    ]),
    client.batch(fakeRequest, [
      {
        action: 'partialUpdateObjectNoCreate',
        indexName: 'cool-index',
        body: {},
      },
    ]),
    client.batch(fakeRequest, [
      {
        action: 'deleteObject',
        indexName: 'cool-index',
        body: {},
      },
    ]),
    client.batch(fakeRequest, [
      {
        action: 'delete',
        indexName: 'cool-index',
      },
    ]),
    client.batch(fakeRequest, [
      {
        action: 'clear',
        indexName: 'cool-index',
      },
    ]),
  ];

  requests.map(req => expect(req).toMatchSnapshot());
});

it('search', () => {
  const fakeRequest = o => o;
  const client = initClient(validParams);
  const request = client.search(fakeRequest, [
    {
      indexName: 'some_index',
      params: {},
    },
  ]);
  expect(request).toMatchSnapshot();
});

it('getLogs', () => {
  const fakeRequest = o => o;
  const client = initClient(validParams);
  const requests = [
    client.getLogs(fakeRequest),
    client.getLogs(fakeRequest, {}),
    client.getLogs(fakeRequest, { offset: 4 }),
    client.getLogs(fakeRequest, { length: 50 }),
    client.getLogs(fakeRequest, { offset: 4, length: 50 }),
  ];

  requests.map(req => expect(req).toMatchSnapshot());
});

it('listIndexes', () => {
  const fakeRequest = o => o;
  const client = initClient(validParams);
  const request = client.listIndexes(fakeRequest);
  const requestWithPage = client.listIndexes(fakeRequest, 1);

  expect(request).toMatchSnapshot();
  expect(requestWithPage).toMatchSnapshot();
});
