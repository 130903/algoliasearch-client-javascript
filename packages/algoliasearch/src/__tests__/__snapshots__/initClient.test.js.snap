// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`batch 1`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "addObject",
        "body": Object {
          "function": [Function],
          "nested": Object {
            "is": Object {
              "all": "fine",
            },
          },
          "some": "data",
        },
        "indexName": "cool-index",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/batch",
}
`;

exports[`batch 2`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "updateObject",
        "body": Object {
          "and": "this is everything now",
          "objectID": "some-cool-object",
          "some": "other data",
        },
        "indexName": "uncool-index",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/batch",
}
`;

exports[`batch 3`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "partialUpdateObject",
        "body": Object {
          "objectID": "some-cool-object",
          "some": "data",
          "the": "rest is still here",
        },
        "indexName": "cool-index",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/batch",
}
`;

exports[`batch 4`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "partialUpdateObjectNoCreate",
        "body": Object {},
        "indexName": "cool-index",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/batch",
}
`;

exports[`batch 5`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "deleteObject",
        "body": Object {},
        "indexName": "cool-index",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/batch",
}
`;

exports[`batch 6`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "delete",
        "indexName": "cool-index",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/batch",
}
`;

exports[`batch 7`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "clear",
        "indexName": "cool-index",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/batch",
}
`;

exports[`contains the correct methods 1`] = `
Array [
  "batch",
  "getLogs",
  "listIndexes",
  "search",
]
`;

exports[`getLogs 1`] = `
Object {
  "method": "GET",
  "path": "/1/logs",
  "qs": Object {
    "length": 10,
    "offset": 0,
  },
}
`;

exports[`getLogs 2`] = `
Object {
  "method": "GET",
  "path": "/1/logs",
  "qs": Object {
    "length": 10,
    "offset": 0,
  },
}
`;

exports[`getLogs 3`] = `
Object {
  "method": "GET",
  "path": "/1/logs",
  "qs": Object {
    "length": 10,
    "offset": 4,
  },
}
`;

exports[`getLogs 4`] = `
Object {
  "method": "GET",
  "path": "/1/logs",
  "qs": Object {
    "length": 50,
    "offset": 0,
  },
}
`;

exports[`getLogs 5`] = `
Object {
  "method": "GET",
  "path": "/1/logs",
  "qs": Object {
    "length": 50,
    "offset": 4,
  },
}
`;

exports[`listIndexes 1`] = `
Object {
  "method": "GET",
  "path": "/1/indexes",
  "qs": undefined,
}
`;

exports[`listIndexes 2`] = `
Object {
  "method": "GET",
  "path": "/1/indexes",
  "qs": 1,
}
`;

exports[`search 1`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "indexName": "some_index",
        "params": Object {},
      },
    ],
  },
  "forceReadHosts": true,
  "method": "POST",
  "path": "/1/indexes/*/queries",
}
`;
